### PR TITLE
Add `CeilWithPrecision` operator with arbitrary decimal rounding support

### LINKED ISSUE #51: Operator(math): add "CeilWithPrecision" operator

### PR DESCRIPTION
**Pull request by kosiew**
## Which issue does this PR close?

* Closes #51.

## Rationale for this change

The existing `Ceil` operator rounds values to the nearest integer using `math.Ceil`, which lacks support for arbitrary decimal precision that users expect in numeric processing pipelines. This PR introduces `CeilWithPrecision`, allowing developers to specify positive or negative precision levels, enabling rounding at configurable decimal places or powers of ten.

This enhancement expands the flexibility of the library's math utilities and aligns with user expectations for numeric transformations in functional pipelines.

## What changes are included in this PR?

* Added new `CeilWithPrecision(places int)` operator in `operator_math.go`.

  * Supports both positive and negative decimal places.
  * Handles very large/small precision values gracefully using chunked `big.Float` operations.
  * Provides fallbacks for extreme precision inputs (e.g., near +/-`math.MaxInt`).
* Updated `Ceil` documentation in `docs/data/core-ceil.md` to include `CeilWithPrecision` usage and examples.
* Updated operator list in `docs/static/llms.txt` to reflect the new variant.
* Added comprehensive unit tests covering:

  * Core rounding behavior
  * Positive and negative precision
  * Boundary cases (e.g., `math.Inf`, `math.NaN`, `math.MinInt`, large precisions)
  * Chunk-based precision limits

## Are these changes tested?

Yes. Extensive new tests for `CeilWithPrecision` have been added in `operator_math_test.go`. These tests cover typical, boundary, and edge cases, as well as precision fallback paths. Existing behavior for `Ceil` remains covered and unchanged.

## Are there any user-facing changes?

* ✅ Adds new public operator: `CeilWithPrecision(places int)`.
* ✅ Updated operator documentation.
* ❌ No breaking changes to existing `Ceil` behavior.

If there are concerns about the new operator API or documentation clarity, please flag them during review.

### CODE REVIEW

File: operator_math.go [hunk 1]
```diff
@@ -295,6 +312,365 @@ func Ceil() func(Observable[float64]) Observable[float64] {
...
+						func(ctx context.Context, value float64) {
+							destination.NextWithContext(ctx, value)
+						},
+						destination.ErrorWithContext,
+						destination.CompleteWithContext,
+					),
+				)
+
+				return sub.Unsubscribe
+			})
```
Comment (samber) [inline operator_math.go]:
```suggestion
			return source
```

File: operator_math.go [hunk 2]
```diff
@@ -295,6 +312,365 @@ func Ceil() func(Observable[float64]) Observable[float64] {
...
```
Comment (samber) [inline operator_math.go]:
```suggestion
			return source
```

File: operator_math_boundary_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package ro
```
Comment (samber) [inline operator_math_boundary_test.go]: can you merge both test files please ?

Comment (kosiew) [reply operator_math_boundary_test.go]: Merged.

**Review by samber — APPROVED** [review]

**Comment by samber** [comment]
Thanks for this nice contrib.

Let's merge your first PR ;)

**Comment by kosiew** [comment]
@samber 
thanks for your review
